lib/src/repository.rs

```diff
@@ -27,8 +27,40 @@ pub trait RepositoryExt {
     /// For a given message of an issue, find the initial message.
     ///
     fn find_tree_init<'a>(&'a self, commit: Commit<'a>) -> Result<Commit>;
+
+    /// Get issue hashes for a prefix
+    ///
+    /// This function returns all known issues known to the DIT repo under the
+    /// prefix provided (e.g. all issues for which refs exist under
+    /// `<prefix>/dit/`). Provide "refs" as the prefix to get only local issues.
+    ///
+    fn get_issue_hashes(&self, prefix: &str) -> Result<Vec<Oid>>;
+
+    /// Get all issue hashes
+    ///
+    /// This function returns all known issues known to the DIT repo.
+    ///
+    fn get_all_issue_hashes(&self) -> Result<Vec<Oid>>;
+}
+
+
+/// Get issue hashes from head references
+///
+/// Retrieve the issue hashes from head references provided. References which
+/// are not head references are ignores. However, the function does not check
+/// whether the references are, in fact, references in a dit namespace.
+///
+fn _head_refs_to_issues(references: References) -> Vec<Oid> {
+    references.names()
+              .filter_map(|name| name.ok())
```

Comment:
Forwarding an error here is pretty cumbersome. For such things, we would want an iterator extension which allows aborting the iteration in some way.